sdk/communication/Azure.Communication.CallingServer/src/CallRecording.cs

```diff
@@ -47,26 +49,32 @@ protected CallRecording()
         /// <summary>
         /// Start recording of the call.
         /// </summary>
-        /// <param name="callLocator"> The callLocator. </param>
-        /// <param name="recordingStateCallbackEndpoint">The uri to send state change callbacks.</param>
-        /// <param name="options">Optional parameters</param>
+        /// <param name="options">Options for start recording</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual Response<RecordingStatusResult> StartRecording(CallLocator callLocator, Uri recordingStateCallbackEndpoint, StartRecordingOptions options = default, CancellationToken cancellationToken = default)
+        public virtual Response<RecordingStatusResult> StartRecording(StartRecordingOptions options, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallRecording)}.{nameof(StartRecording)}");
             scope.Start();
             try
             {
-                StartCallRecordingRequestInternal request = new StartCallRecordingRequestInternal(CallLocatorSerializer.Serialize(callLocator))
+                StartCallRecordingRequestInternal request = new(CallLocatorSerializer.Serialize(options?.CallLocator))
```

Comment:
options cannot be optional, we should add a check for it.

Reply:
Added a check